collector/collector.go

```diff
@@ -20,94 +20,44 @@ import (
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sync"
-	"time"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // PipelineServiceCollector struct
 type PipelineServiceCollector struct {
 	logger            log.Logger
-	mutex             sync.Mutex
 	durationScheduled *prometheus.GaugeVec
 	durationCompleted *prometheus.GaugeVec
-	pipelineRuns      map[string]*v1beta1.PipelineRun
-}
 
-var completedDuration float64
-var scheduledDuration float64
+	client client.Client
+}
 
-func NewCollector(logger log.Logger) (*PipelineServiceCollector, error) {
+func NewCollector(logger log.Logger, client client.Client) (*PipelineServiceCollector, error) {
+	//TODO should this be converted to a Desc so we can use constant metrics
 	durationScheduled := prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Name: "pipelinerun_duration_scheduled_seconds",
 		Help: "Duration in seconds for a PipelineRun to be scheduled.",
+		//TODO the uid will get us a unique entry, but wouldn't adding namespace help usability with humans consuming this metric?
```

Comment:
How are we not exploding the metrics stack??? If we need data on how long an individual pipeline ran, we should get it from Results.

Reply:
Yes my comment did not go that far, but in my head at least ;-), I am on the same page with you wrt the fact the labeling should change here, from both usability as well as scalability, hence why I was calling it out.

As I trid to alude to in this PR's description, I would not be suprised if the way the metrics are defined also contributes to the exporter instability we are seeing in prod etc.  Just switching to the caching client vs. non caching client will help prove that out.

Reply:
I'll update the TODO to make sure the scalability concern is clear.  And I'll go ahead and open the story off the current epic to get into this.

Reply:
story https://issues.redhat.com/browse/PLNSRVCE-1310 opened